app/Models/EntryDAO.php

```diff
@@ -27,6 +27,34 @@ public static function sqlIgnoreConflict(string $sql): string {
 		return str_replace('INSERT INTO ', 'INSERT IGNORE INTO ', $sql);
 	}
 
+	/** @return array{pattern?:string,matchType?:string} */
+	protected static function regexToSql(string $regex): array {
+		if (preg_match('#^/(?P<pattern>.*)/(?P<matchType>[im]*)$#', $regex, $matches)) {
+			return $matches;
+		}
+		return [];
+	}
+
+	/** @param array<int|string> $values */
+	protected static function sqlRegex(string $expression, string $regex, array &$values): string {
+		$matches = static::regexToSql($regex);
+		if (isset($matches['pattern'])) {
+			$matchType = $matches['matchType'] ?? '';
+			if (!str_contains($matchType, 'i')) {
+				// Case-sensitive matching
+				$matchType .= 'c';
+			}
+			$values[] = $matches['pattern'];
```

Comment:
Looks like you haven't implemented multi-line matching for MySQL yet. Just adding this comment to track when it's completed.

Reply:
It looks like multiline mode is only available with MariaDB, not MySQL https://mariadb.com/kb/en/regexp/
MariaDB has indeed a new PCRE engine https://mariadb.com/kb/en/pcre/

Reply:
Implemented